mammoth/trainer.py

```diff
@@ -476,6 +493,9 @@ def _gradient_accumulation(
 
             try:
                 if loss is not None:
+                    if torch.isnan(loss):
+                        raise Exception('Loss blowout')
```

Comment:
can we/should we type this?

Reply:
I think typed exceptions are mainly useful for recoverable problems that you may want to catch somewhere higher up. Blowing out of the training loop is not going to be recoverable.

Reply:
I express a strong disagreement.

1. You do catch it (and everything else) on line 519. 
2. If you want a clean break, then make it a break / branching out cond. If you want a weird stack jump behavior, make it clear to the other devs you're doing a custom hack around a potential problem 
3. I would also argue typing helps  other devs: e.g., assume my dev crashes because of a typo producing a out of bounds, not being caught by this try/except here is bound to make my stack less of pain

The minimal thing is to just make it a runtime so as to avoid cases as in 3, but i wouldn't be against finding something more exotic / custom, that way you could except ExoticError to deal with nan blowouts and except RuntimeError for cuda ooms

Reply:
You are absolutely right.

I already forgot what I was trying to do here. The catch-everything-and-retry-forever was already there from before, and I tried to make it less bad, when I should have just removed it.

I'll remove the try/except, and the `--max_nan_batches` opt.